### PR TITLE
Added Longest Palindromic Subsequence Under Dynamic Programming in C-Plus-Plus

### PR DESCRIPTION
**Pull request by VA-ibh-AV**
…-Plus

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Find Longest Palindromic Subsequence
Can be solved in O(N^2) by using Bottom-Up Dynamic Programming

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

### CODE REVIEW

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Few changes: 

- Use Comments above the Line of Code not beside it
- Add a new line at the EOF 
- Run a formatter across your Code and indent it properly

**Comment by VA-ibh-AV** [comment]
@HarshCasper I had made the required changes, can you please check it.

File: C-Plus-Plus/dp/longest_palindromic_subsequence.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+int main() {
+
+	// Input The String
+	string s;
+	cin >> s;
+
+	cout << "Longest Palindrmoic Subsequence of " << s << " is: " << lps(s) << '\n';
+
+	/*
+	    Try Running with input
```
Comment (atarax665) [inline C-Plus-Plus/dp/longest_palindromic_subsequence.cpp]: Shift sample I/O at the end of the program and also add time and space complexity of your program.

File: C-Plus-Plus/dp/longest_palindromic_subsequence.cpp [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	}
+
+	// i.e the whole string ex: abba, dp[0][3] = 4
+	return dp[0][n - 1];
+}
+
+int main() {
+
+	// Input The String
+	string s;
```
Comment (atarax665) [inline C-Plus-Plus/dp/longest_palindromic_subsequence.cpp:67]: Add a statement to prompt user to enter the string.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by VA-ibh-AV** [comment]
@atarax665 I had made the changes, can you please check it 😊😊

File: C-Plus-Plus/dp/longest_palindromic_subsequence.cpp [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+		Space Complexity O(N^2)
+
+	    Sample Input:
+	    s = "vavbhhb"
+		Output: 4
+	    Explanation:
+	    -> there are two palindromic subsequences vav, bhhb
+	    -> longest is bhhb therefore the ans is 4
+	*/
+
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/longest_palindromic_subsequence.cpp:89]: Left-align this and remove unnecessary lines of code

File: C-Plus-Plus/dp/longest_palindromic_subsequence.cpp [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        if let's say "agbdba", s[i] = s[j] i.e s[0] == s[5]
+        then ans = 2 + lps(i+1,j-1) i.e check the ans for "gbdb"
+        why 2 ??
+        because let's say for "agbdba" first and last are same "a|gbdb|a"
+        the length should be adding those charcters plus LPS of "gbdb"
+
+        for "gbdb" s[i] != s[j] i.e s[0] != s[3]
+        ans = max(lps(i+1,j),lps(i,j-1)) i.e max lps for "gbd" and "bdb"
+
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/longest_palindromic_subsequence.cpp:29]: Approach should be captured through Code. Remove it here

File: C-Plus-Plus/dp/longest_palindromic_subsequence.cpp [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        -> A subsequence is a sequence that can be derived from another sequence
+        by deleting some or no elements without changing the order of the remaining elements.
+
+    Example
+        s = "bbbab"
+        ans = 4, "bbbb" is the longest palindromic subsequence
+
+        s = "agbdba"
+        ans = 5, "abdba" is the longest palindromic subsequence
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/longest_palindromic_subsequence.cpp:14]: Sample I/O are good enough for the example

File: C-Plus-Plus/dp/longest_palindromic_subsequence.cpp [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+	}
+
+	// i.e the whole string ex: abba, dp[0][3] = 4
+	return dp[0][n - 1];
+}
+
+int main() {
+
+	// Input The String
+	string s;
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/longest_palindromic_subsequence.cpp:67]: Use meaningful variable names

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by atarax665** [comment]
Hey @BullHoN Are you still working on this?

**Pull request closed without merging**